packages/@holaplexui-playground/hooks/useUpload.ts

```diff
@@ -0,0 +1,37 @@
+import { useState } from 'react';
+import {
+  DropzoneInputProps,
+  DropzoneRootProps,
+  useDropzone
+} from 'react-dropzone';
+
+interface UploadContext {
+  getInputProps: <T extends DropzoneInputProps>(props?: T | undefined) => T;
+  getRootProps: <T extends DropzoneRootProps>(props?: T | undefined) => T;
+  isDragActive: boolean;
+  droppedFiles: File[] | null;
+}
+
+export default function useUpload(): UploadContext {
+  const [droppedFiles, setDroppedFiles] = useState<File[] | null>(null);
```

Comment:
You don't need state looking at the docs you can pass an onChange to the getInputProps({ onChange }) don't pass through the hook. Pass to the DragDrop component.